soc/arm/st_stm32/stm32f7/soc.c

```diff
@@ -35,12 +35,12 @@ static int st_stm32f7_init(const struct device *arg)
 	LL_FLASH_EnableART();
 
 	key = irq_lock();
-
+#ifndef CONFIG_NOCACHE_MEMORY
```

Comment:
Not the right solution. Let's remove F7 case from this PR.

Reply:
removing the nucleo_f746zg